portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java

```diff
@@ -618,5 +647,181 @@ public String getSkinPrefix() {
 		return "";
 	}
 
+	public List<BullhornAlert> getSocialAlerts(String userId) {
+
+		List<BullhornAlert> alerts = sqlService.dbRead(
+                "SELECT * FROM SOCIAL_ALERTS WHERE TO_USER = ? ORDER BY EVENT_DATE DESC"
```

Comment:
I thought it made more sense there as it is immediately apparent what the query is. I find it a bit of a pain when queries are moved hundreds of lines of code away from where they are used. The compiler should optimise that string into a constant so it shouldn't be a performance problem.